src/robotFactory.js

```diff
@@ -1,10 +1,106 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords = {
+      x: 0,
+      y: 0,
+    },
```

Comment:
```suggestion
    coords,
```
Unnecessary, you do this when destructurize coords.

Reply:
and if coords isn't an object?

Reply:
Of course, it can be anything. But what if I say that `coords.x` can possibly be not a number? Or cargo can possibly be a string and doesn't have the weight prop.
That is why we are using Typescript 😉
In this task, we can be sure that coords are always objects to simplify checks.